src/cata_tiles.cpp

```diff
@@ -2481,7 +2496,12 @@ bool cata_tiles::draw_vpart( const tripoint &p, lit_level ll, int &height_3d )
     bool draw_highlight = cargopart && !veh->get_items( cargopart->part_index() ).empty();
 
     if( !veh->forward_velocity() ) {
-        g->u.memorize_tile( g->m.getabs( p ), vpid, subtile, veh_dir );
+        std::bitset<SEEX *MAPSIZE *SEEY *MAPSIZE> &memory_seen_cache =
+            g->m.get_cache( p.z ).map_memory_seen_cache;
+        if( !memory_seen_cache[ static_cast<size_t>( p.x + ( p.y * SEEY * MAPSIZE ) ) ] ) {
+            memory_seen_cache.set( static_cast<size_t>( p.x + ( p.y * SEEY * MAPSIZE ) ) );
+            g->u.memorize_tile( g->m.getabs( p ), vpid, subtile, veh_dir );
+        }
```

Comment:
Each of the four chunks added to this file are almost identical.  Maybe factor them out into a function?

Reply:
Good point, I'm going to push this logic down into a map helper. (Better name desperately needed)
This also avoids scattering around the indexing logic.
Caller
```
if( check_and_set_seen_cache( p ) ) {
    g->u.memorize_tile( ... );
}
```
Helper
```
bool map::check_and_set_seen_cache( const tripoint &p ) {
    // get cache reference
    if( !memory_seen_cache[ static_cast<size_t>( p.x + ( p.y * SEEY * MAPSIZE ) ) ] ) {
        memory_seen_cache.set( static_cast<size_t>( p.x + ( p.y * SEEY * MAPSIZE ) ) );
        return true;
    }
    return false;
}
```